apps/rxjs-race-condition/src/app/app.component.ts

```diff
@@ -1,37 +1,33 @@
-import {
-  ChangeDetectionStrategy,
-  Component,
-  inject,
-  OnInit,
-} from '@angular/core';
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
-import { take } from 'rxjs';
+import { Observable, take } from 'rxjs';
 import { TopicModalComponent } from './topic-dialog.component';
 import { TopicService, TopicType } from './topic.service';
+import { AsyncPipe, NgIf } from '@angular/common';
 
 @Component({
   standalone: true,
   selector: 'app-root',
-  template: ` <button (click)="openTopicModal()">Open Topic</button> `,
+  template: `
+    <button *ngIf="topics$ | async as topics" (click)="openTopicModal(topics)">
+      Open Topic
+    </button>
+  `,
+  imports: [NgIf, AsyncPipe],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AppComponent implements OnInit {
+export class AppComponent {
   title = 'rxjs-race-condition';
   dialog = inject(MatDialog);
   topicService = inject(TopicService);
-  topics: TopicType[] = [];
+  topics$: Observable<TopicType[]> = this.topicService
+    .fakeGetHttpTopic()
+    .pipe(take(1));
```

Comment:
question: why did you use a take 1 here ?

Reply:
I forgot to delete the pipe as we don't need it. just pushed new changes